src/maestral/main.py

```diff
@@ -865,6 +882,99 @@ def list_revisions(self, dbx_path: str, limit: int = 10) -> List[StoneType]:
 
         return entries
 
+    def get_file_diff(
+        self, dbx_path: str, old_rev: str, new_rev: Optional[str] = None
+    ) -> List[str]:
+        """
+        Download the two revisions if necessary and create a
+        diff with the python difflib library. If new_rev is None,
+        it will compare the old revision to the current file.
+
+        :param old_rev: Hash of old revision.
+        :param new_rev: Hash of new revision.
+        :returns: Diff as a list of lines.
+        :raises UnsupportedFileTypeForDiff: if file type is not supported.
+        :raises UnsupportedFileTypeForDiff: if file content could not be decoded.
+        :raises MaestralApiError: if file could not be read for any other reason.
+        """
+
+        def pretty_date(s: datetime) -> str:
+            """
+            Prettify the 'client_modified' metadata.
+            """
+
+            return s.astimezone().strftime("%d %b %Y at %H:%M")
+
+        def download_rev(f: IO[bytes], rev: str) -> Tuple[str, str]:
+            """
+            Download a rev to 'f' and return the location and the date.
+            """
+
+            location = f.name
+            date = pretty_date(
+                self.client.download(dbx_path, location, rev=rev).client_modified
+            )
+            return (location, date)
```

Comment:
I would probably move the temp file handling to this function and return the content directly. This would enable you to write:

```Python
with tempfile.TemporaryFile(... ) as f:
    # perform download and read content
```

Reply:
Looks a lot cleaner now.